Src/Sarif.PatternMatcher/AnalyzeCommand.cs

```diff
@@ -76,9 +76,10 @@ public AnalyzeCommand(IFileSystem fileSystem = null)
                     FileVersionInfo fvi = fileSystem.FileVersionInfoGetVersionInfo(Path.Combine(directory, definitions.ValidatorsAssemblyName));
 
                     name = $"{fvi?.CompanyName}/{fvi?.FileDescription}/{name}";
+
                     // TBD add version details. Breaks test baselines currently.
-                    //semanticVersion = fvi?.ProductVersion;
-                    //version = fvi?.FileVersion;
+                    // semanticVersion = fvi?.ProductVersion;
```

Comment:
>semanticVersion [](http://example.com/codeflow?start=23&length=15)

So here we've made a code clean-up while ignoring  something important, a dev (me! eep) has left a work item undone. Far more important than injecting spaces is that devs look at this and either update the code comment or create a tracking issue for the work.

Welcome to code clean-up. :) This will rub your face in all our technical debt. We perform code clean-up not merely to make symptoms (warnings) for away, we do it to rub our faces in our debt and potentially drive clean-up.

Please file an issue for this open item, which is actually pretty important TODO.

Reply:
New user story created:

[Add version details to ToolComponent in CreateSkimmers](https://dev.azure.com/mseng/1ES/_workitems/edit/2066916)

Reply:
There are about 50 TODOs and about 13 TBDs in BinSkim.

Reply:
@schlaman-ms Thanks for opening the work item. But I think there is a subtlety here. To me, the value of work items come from the fact that it forms a connected component. For the work item you created, there are the following issues:

1. If someone is browsing the code and reaches this TODO, there is no way to click and jump to the corresponding work item.
2. If someone is browsing our ADO board and reaches the work item, there is no way to point back to this code.
3. Since this is open-source, and we drop in an ADO work-item link, it will be broken for most people. 

We should think of a coding standard for this as well:

1. For Todos in open-source, raise issues. Ensure the issue is linked from code, and there are links to the code from the issue.
2. For internal projects, replace "issues" with ADO work items.

@michaelcfanning what do you think about this suggestion?